src/System.Management.Automation/engine/debugger/debugger.cs

```diff
@@ -4160,7 +4160,24 @@ internal void TraceVariableSet(string varName, object value)
             // because 'ToStringParser' would iterate through the enumerator to get the individual elements, which will
             // make irreversible changes to the enumerator.
             bool isValueAnIEnumerator = PSObject.Base(value) is IEnumerator;
```

Comment:
Unless this condition is reused elsewhere, it seems like we could just embed it into the `if (...)` condition directly, rather than assign it to a variable

Reply:
It would be unrelated change.